command-expansion-server/src/getActivityTypescript.ts

```diff
@@ -0,0 +1,131 @@
+import {GraphQLClient, gql} from 'graphql-request';
+import {globalDeclaration, indent, interfaceDeclaration} from "./packages/lib/CodegenHelpers.js";
+import {ErrorWithStatusCode} from "./utils/ErrorWithStatusCode.js";
+
+enum SchemaTypes {
+  Int = 'int',
+  Real = 'real',
+  Duration = 'duration',
+  Boolean = 'boolean',
+  String = 'string',
+  Series = 'series',
+  Struct = 'struct',
+  Variant = 'variant',
+}
+
+type Schema =
+  | IntSchema
+  | RealSchema
+  | DurationSchema
+  | BooleanSchema
+  | StringSchema
+  | SeriesSchema
+  | StructSchema
+  | VariantSchema
+  ;
+
+interface BaseSchema<T extends SchemaTypes | unknown> {
+  type: T,
+}
+
+type StringSchema = BaseSchema<SchemaTypes.String>;
+type BooleanSchema = BaseSchema<SchemaTypes.Boolean>;
+type IntSchema = BaseSchema<SchemaTypes.Int>;
+type RealSchema = BaseSchema<SchemaTypes.Real>;
+type DurationSchema = BaseSchema<SchemaTypes.Duration>;
+
+interface SeriesSchema extends BaseSchema<SchemaTypes.Series> {
+  items: Schema,
+}
+
+interface StructSchema extends BaseSchema<SchemaTypes.Struct> {
+  items: {
+    [key: string]: Schema,
+  }
+}
+
+interface VariantSchema extends BaseSchema<SchemaTypes.Variant> {
+  variants: {
+    key: string,
+    label: string,
+  }[]
+}
+
+interface GraphQLActivityParameter {
+  order: number,
+  schema: Schema,
+}
+
+interface GraphQLActivity {
+  name: string,
+  parameters: GraphQLActivityParameter[],
+  requiredParameters: string[],
+}
+
+export async function getActivityTypescript(
+  graphqlClient: GraphQLClient,
+  missionModelId: number,
+  activityTypeName: string
+): Promise<string> {
+  console.log(`query parameters from ${activityTypeName} with missionModelId: ${missionModelId}`);
+  const response = await graphqlClient.request<{
+    activity_type: GraphQLActivity[]
+  }>(gql`
+    query GetParameters {
+      activity_type(where: {
+        model_id: { _eq: ${missionModelId} },
+        name: { _eq: ${activityTypeName} },
+      }) {
+        name
+        parameters
+      }
+    }
+  `);
```

Comment:
🔻 Would it be possible to use GraphQL variables instead of string interpolation here? It would be a very minimal change, and it would prevent GraphQL injection issues via the activityTypeName:
```suggestion
  }>(gql`
    query GetParameters($missionModelId: Int!, $activityTypeName: String!) {
      activity_type(where: {
        model_id: { _eq: $missionModelId },
        name: { _eq: $activityTypeName },
      }) {
        name
        parameters
      }
    }
  `, {missionModelId, activityTypeName});
```

Reply:
Thanks for the suggestion. I made the change and verified it works as expected